browser/src/components/NodeSummary.tsx

```diff
@@ -171,7 +159,10 @@ export function NodeSummary({ graphManager, weightService, componentName, nodeNa
           <NodeLink
               node={node}
               weight={weightService.getWeight(componentName, node.key)}
-          />
+              onSelect={() => 
+                {!fullDetails && history.push(Routes.GraphNode(componentName, node.key))}
```

Comment:
It's ok to navigate to itself in the `fullDetails=true` mode, let's remove the `fullDetails` check here to make logic simple